python/lsst/daf/butler/core/repoRelocation.py

```diff
@@ -64,8 +64,12 @@ def replaceRoot(configRoot: str, butlerRoot: Optional[ResourcePathExpression]) -
     ValueError
         Raised if ``butlerRoot`` is not set but a value is required.
     """
-    # Do nothing if there is nothing to be done
-    if BUTLER_ROOT_TAG not in configRoot:
+    # Do nothing if there is nothing to be done or if there is a type
+    # mismatch.
+    try:
+        if BUTLER_ROOT_TAG not in configRoot:
+            return configRoot
+    except TypeError:
```

Comment:
Funny you should say that. I had to add the try/except because I'm injecting the `TestClient` into the `Config` to override the db field after I have read it and the `Config` classes are not typed at all so eventually what is normally a string ends up being a `TestClient` and the code blows up. Ordinarily the db would be a URL to the server but in the test code we don't run up a server and so this work around has to be used.

Reply:
Mypy should have complained about it, because `configRoot` is annotated as `str` and that cannot raise TypeError 🙂